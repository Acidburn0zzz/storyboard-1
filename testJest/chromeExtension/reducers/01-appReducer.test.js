/* eslint-env jest */
/* eslint-disable no-extend-native */
import reducer from '../../../lib/chromeExtension/reducers/appReducer';

describe('appReducer', () => {
  it('should have correct initial state', () => {
    Date.prototype.getTime = jest.genMockFunction().mockReturnValue(0);
    const state = reducer(undefined, { type: '' });
    expect(state).toMatchSnapshot();
  });

  it('should process a simple action', () => {
    let state = reducer(undefined, { type: '' });
    state = reducer(state, { type: 'CX_CONNECTED' });
    expect(state.cx.cxState).toBe('CONNECTED');
  });
});
