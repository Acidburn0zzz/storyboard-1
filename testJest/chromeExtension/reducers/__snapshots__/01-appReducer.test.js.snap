exports[`appReducer should have correct initial state 1`] = `
Object {
  "cx": Object {
    "cxState": "DISCONNECTED",
    "fLoginRequired": null,
    "localClientFilter": null,
    "login": null,
    "loginState": "LOGGED_OUT",
    "serverFilter": null,
    "wsState": "DISCONNECTED",
  },
  "settings": Object {
    "colorClientBg": "aliceblue",
    "colorClientBgIsDark": false,
    "colorClientFg": "black",
    "colorServerBg": "rgb(214, 236, 255)",
    "colorServerBgIsDark": false,
    "colorServerFg": "black",
    "colorUiBg": "white",
    "colorUiBgIsDark": false,
    "colorUiFg": "rgb(64, 64, 64)",
    "fCollapseAllNewStories": false,
    "fDiscardRemoteClientLogs": false,
    "fExpandAllNewAttachments": false,
    "fShorthandForDuplicates": true,
    "fShowClosedActions": false,
    "forgetHysteresis": 0.25,
    "maxRecords": 800,
    "timeType": "LOCAL",
  },
  "stories": Object {
    "closedStories": Object {},
    "localHubId": undefined,
    "mainStory": Object {
      "fExpanded": true,
      "fHierarchical": true,
      "fOpen": true,
      "fWrapper": true,
      "records": Array [
        Object {
          "fExpanded": true,
          "fHierarchical": true,
          "fMain": true,
          "fOpen": true,
          "fServer": false,
          "fStoryObject": true,
          "id": "main_0",
          "lastAction": "CREATED",
          "numRecords": 0,
          "pathStr": "records/0",
          "records": Array [],
          "src": "main",
          "status": undefined,
          "storyId": "main_0",
          "t": 0,
          "title": "Client",
        },
        Object {
          "fExpanded": true,
          "fHierarchical": true,
          "fMain": true,
          "fOpen": true,
          "fServer": true,
          "fStoryObject": true,
          "id": "main_1",
          "lastAction": "CREATED",
          "numRecords": 0,
          "pathStr": "records/1",
          "records": Array [],
          "src": "main",
          "status": undefined,
          "storyId": "main_1",
          "t": 0,
          "title": "Server",
        },
      ],
    },
    "openStories": Object {},
    "quickFind": "",
  },
}
`;
